File: 997-Find the Town Judge.java
//Time complexity=O(N)
//Space complexity=O(N)
//Initially we are creating an array of length equal to the number of people. And then we are traversing the entire array of trust to check the dependency of each judge on each other. If there is an out-degree then we are decrementing the count and if there is in-degree then we are incrementing, in the array we created. The index at which the value= N-1 is our answer.    
class Solution {
    public int findJudge(int N, int[][] trust) {
        
        
        int[] trustArr = new int[N];
        
        for(int[] arr : trust)
        {
            trustArr[arr[0]-1]--;
            trustArr[arr[1]-1]++;
        }
        
        for(int i = 0 ; i < trustArr.length ; i ++)
        {
            if(trustArr[i] == N-1)
            {
                return i+1;
            }
        }
        
        return -1;
    }
}

File: 997. Find the Town Judge.py
#Time Complexity:O(n)
#Space Complexity:O(n)

class Solution(object):
    def findJudge(self, n, trust):
        """
        :type n: int
        :type trust: List[List[int]]
        :rtype: int
        """
        incoming=[0]*(n)
        outgoing=[0]*(n)
        for node in trust:
                outgoing[node[0]-1]=outgoing[node[0]-1]+1
                incoming[node[1]-1]=incoming[node[1]-1]+1
        print(outgoing)
        print(incoming)
        for i in range(len(outgoing)):
            if outgoing[i]==0 and incoming[i]==n-1:
                return i+1
        return -1